### PR TITLE
Decoding raw ByteString TokenNames

### PR DESCRIPTION
**Pull request by gege251**
In a project where we use bot-plutus-interface, we use a non utf8 tokenname (it is a hash of a few things), so the current implementation would fail.

It turned out that we don't need to explicitly decode the tokenname from hex to a utf8 ByteString, it can be decoded straight into bs and it would still be interpreted the same way inside the Plutus code.
The cli adds a 0x prefix for raw bytestring tokennames, but we can simply skip over that.

### CODE REVIEW

**Review by t4ccer — APPROVED** [review]

**Comment by samuelWilliams99** [comment]
Bit confused over the cli logic here - does it just add a `0x` if the hex after it doesn't decode to utf8?
does the existance of the `0x` actually change the way the name is encoded? I assume that - given we decode to bytestring, hex would always be enough to encode the name

File: src/BotPlutusInterface/UtxoParser.hs [hunk 1]
```diff
@@ -106,7 +108,8 @@ tokenNameParser = do
   where
     tokenName = do
       void $ char '.'
-      Value.tokenName . encodeUtf8 <$> decodeHex (takeWhile (/= ' '))
+      void (string "0x") <|> pure ()
```
Comment (samuelWilliams99) [inline src/BotPlutusInterface/UtxoParser.hs:111]: Can maybe use optional here
```suggestion
      void $ optional $ string "0x"
```

**Review by samuelWilliams99 — APPROVED** [review]

**Comment by gege251** [comment]
@samuelWilliams99 Couldn't find any documentation about this, but it seems like that is the case: if the token name is not a utf8 encoded string, then an 0x will be added to it. But maybe @t4ccer knows more about it, he bumped into this issue.

I spent some time to track down where the `0x` is added in the cardano-cli source code, but couldn't find it TBH. This is where I ended up:
https://github.com/input-output-hk/cardano-node/blob/814df2c146f5d56f8c35a681fe75e85b905aed5d/cardano-api/src/Cardano/Api/Value.hs#L404
https://github.com/input-output-hk/cardano-node/blob/814df2c146f5d56f8c35a681fe75e85b905aed5d/cardano-api/src/Cardano/Api/SerialiseRaw.hs#L28-L32

**Comment by samuelWilliams99** [comment]
How odd, yeah there doesn't appear to be any logic adding that - even the serialisation instance for AssetName is just `id`
Either way - if this fixes the issue - fine by me. Any usual hex string couldn't have a `0x` prefix anyway, so its not a breaking change

**Pull request merged**